components/forwarder/forwarder.go

```diff
@@ -0,0 +1,88 @@
+package forwarder
+
+import (
+	"context"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// Forwarder subscribes to the topic provided in the config and publishes them to the destination topic embedded in the enveloped message.
+type Forwarder struct {
+	router    *message.Router
+	publisher message.Publisher
+	logger    watermill.LoggerAdapter
+	config    Config
+}
+
+// NewForwarder creates a forwarder which will subscribe to the topic provided in the provided config using the provided subscriber.
+// It will publish messages received on this subscription to the destination topic embedded in the enveloped message using the provided publisher.
+//
+// Provided subscriber and publisher can be from different Watermill Pub/Sub implementations, i.e. MySQL subscriber and Google Pub/Sub publisher.
+func NewForwarder(subscriberIn message.Subscriber, publisherOut message.Publisher, logger watermill.LoggerAdapter, config Config) (*Forwarder, error) {
+	config.setDefaults()
+
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
```

Comment:
if we are using a router here, then we might want to use some middlewares as well
I'm thinking additional param `routerMiddlewares []Middleware` or additional config field.

We could also pass a router by constructor param, but would have to assert that it's not already running so that we can add the handler.

Reply:
But if we pass a router from outside, then `forwarder.Run()` and `forwarder.Close()` no longer make sense, because the router may be run or closed independently.
Probably better just to create a router in the constructor.

Otherwise, we'd have to turn the constructor into sth like `AddForwarderHandler(subscriberIn Subscriber, publisherOut Publisher, logger LoggerAdapter, config Config, router Router) error`

Reply:
According to our discussion last time, I have added middlewares to the forwarder's config. We will keep router as an internal detail and will not try to mimic its "API".

Reply:
👍